doc/final/final_report.tex

```diff
@@ -81,10 +82,97 @@ \section{Introduction}
+The other interactions happen on the Map Panel. First of all, the user can drag and move the map to see more details of the simulation. Because the size of the map is bigger than the size of the Map Panel, we use this solution to help the user to see the complete map information. Then the user can set or remove the traffic lights just by clicking on the specific junction showed on the map.
+
+\subsubsection{Design and implementation}
+\includegraphics[width=12.5cm]{class.eps}
+\paragraph{Class PanelMouseAction} implements MouseInputListener, which is added on the Map Panel. The main task for this class is processing the requests from the Mouse operation in the Map Panel. The first operation is dragging the map. It is easy to implement the function of dragging the panel, however, it will get more complex if we take the range into consider. The map in our system is a rectangle that has limited length and height. So the range of dragging should be the minimum length and width to maximum length and width. The second one is set or remove the traffic lights. In order to ensure the operability, the selectable range should be enlarged to the whole junction. After clicking the junction, the relevant parameters will be sent back to the "common" package.
+\paragraph{Class ControlPanel} extends JPanel, which is added on the Main Frame. It is responsible for most of the interaction works. It uses the static parameter simulationListener in MainFrame to transfer inputs from the user to the server end mostly. While for the setting of debug mode, it transfers the Boolean parameter to the MapPanel directly. In addition, it should do the necessary checking of inputs before passing the values to the server side. For the invalid inputs, it will generate a JDialog object to prompt the user to make the appropriate changes. As to the control of starting or pausing, clearing of the simulation. It will use stateProviderController to control the main thread of the system.
+\paragraph{Class ControlPanelDocumentListnener} implements DocumentListener, which is added to the text fields in Control Panel. It is used to check the validity of inputs. If the input is invalid, a JDialog that has related information will be generated.
+\paragraph{Class StateProviderController} extends Thread, which is the main thread of the simulation. The thread is generated by the Bootstrapper class, and it keeps sleep state until the user starts the simulation. The Boolean parameter isSleep controlled the starting and pausing of the simulation. When the user chooses to clear the simulation, a new StateProvider will be passed to the simulation.
+\paragraph{Class Bootstrapper} is generated by the main starter of the system. It is used to create the main thread StateProviderController and start it. It also generates the Main Frame and set relevant parameters of the simulation.
```

Comment:
I think it's good to clearly state that Bootstapper and StateProviderController are parts of a different package and the control panel code just has references to them.